src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -114,60 +139,88 @@ void rtc_enable_alarm(void) {
     }
 }
 
-static void rtc_irq_handler(void) {
+void rtc_disable_alarm(void) {
+    // Disable matching and wait for it to stop being active
+    hw_clear_bits(&rtc_hw->irq_setup_0, RTC_IRQ_SETUP_0_MATCH_ENA_BITS);
+    while (rtc_hw->irq_setup_0 & RTC_IRQ_SETUP_0_MATCH_ACTIVE_BITS) {
+        tight_loop_contents();
+    }
+}
+
+static void __no_inline_not_in_flash_func(rtc_irq_handler)(void) {
     // Always disable the alarm to clear the current IRQ.
     // Even if it is a repeatable alarm, we don't want it to keep firing.
     // If it matches on a second it can keep firing for that second.
     rtc_disable_alarm();
 
     if (_alarm_repeats) {
```

Comment:
Should all the `if (_alarm_repeats)` checks actually be `if (_alarm_repeats != NO_REPEAT)` ? :shrug:

Reply:
a question of coding guidelines. Technically it's the same

Reply:
Fair enough, that's a Graham-decision then :slightly_smiling_face: